nltk/tokenize/casual.py

```diff
@@ -0,0 +1,304 @@
+    def __init__(self, preserve_case=True):
+        self.preserve_case = preserve_case
+
+    def tokenize(self, text):
+        """
+        :param s: str
+        :rtype: list(str)
+        :return: a tokenized list of strings; concatenating this list returns
+        the original string if preserve_case=False
+        """
+        # Try to ensure unicode:
+        try:
+            text = str(text)
+        except UnicodeDecodeError:
+            text = str(text).encode('string_escape')
+            text = str(text)
+        # Fix HTML character entities:
+        text = self._html2unicode(text)
+        # Tokenize:
+        words = WORD_RE.findall(text)
+        # Possibly alter the case, but avoid changing emoticons like :D into :d:
+        if not self.preserve_case:
+            words = list(map((lambda x : x if EMOTICON_RE.search(x) else
+                              x.lower()), words))
+        return words
+
+
+    def _html2unicode(self, text):
+        """
+        Try to replace all the HTML entities in `s` with their corresponding
+        Unicode characters.
+        """
+        # First the digits:
+        ents = set(HTML_ENTITY_DIGIT_RE.findall(text))
+        if len(ents) > 0:
+            for ent in ents:
+                entnum = ent[2:-1]
+                try:
+                    entnum = int(entnum)
+                    text = text.replace(ent, chr(entnum))
+                except:
+                    pass
+        # Now the alpha versions:
+        ents = set(HTML_ENTITY_ALPHA_RE.findall(text))
+        ents = [e for e in ents if not e == AMP]
+        for ent in ents:
+            entname = ent[1:-1]
+            try:
+                text = text.replace(ent, chr(htmlentitydefs.name2codepoint[entname]))
+            except:
+                pass
+            text = text.replace(AMP, " and ")
+        return text
+
+######################################################################
+# Tokenization Function
+######################################################################
+
+def casual_tokenize(text, preserve_case=True):
+    """
+    Convenience function for wrapping the tokenizer.
+    """
+    return TweetTokenizer(preserve_case=preserve_case).tokenize(text)
+
+###############################################################################
+
+if __name__ == '__main__':
+    s0 = "This is a cooool #dummysmiley: :-) :-P <3 and some arrows < > -> <--"
+    s1 = "Naps are a must \ud83d\ude34\ud83d\ude34"
+    s2 = "Renato fica com muito medo de ouvir meus \u00e1udios perto da gaja\
+    dele, pois s\u00f3 falo merda KKK"
+    s3 = "\u0412\u043b\u0430\u0434\u0435\u043b\u0435\u0446 20th Century Fox\
+    \u043d\u0430\u043c\u0435\u0440\u0435\u043d\
+    \u043a\u0443\u043f\u0438\u0442\u044c Warner Bros."
+    s4 = "RT @facugambande: Ya por arrancar a grabar !!! #TirenTirenTiren vamoo !!"
+    s5 = "http://t.co/7r8d5bVKyA http://t.co/hZpwZe1uKt\
+    http://t.co/ZKb7GKWocy Ничто так не сближает\
+    людей"
```

Comment:
Note that in Python 2.x all strings are utf8-encoded bytestrings here, so the example doesn't look correct - users should pass unicode to the library. To fix it we can add `from __future__ import unicode_literals` to the top of the file.